service/src/main/java/bio/terra/workspace/amalgam/landingzone/azure/AzureLandingZoneApiController.java

```diff
@@ -0,0 +1,167 @@
+import bio.terra.workspace.service.iam.AuthenticatedUserRequestFactory;
+import bio.terra.workspace.service.iam.SamService;
+import bio.terra.workspace.service.workspace.exceptions.CloudContextRequiredException;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import javax.servlet.http.HttpServletRequest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+
+@Controller
+public class AzureLandingZoneApiController extends ControllerBase implements LandingZonesApi {
+  private static final Logger logger = LoggerFactory.getLogger(AzureLandingZoneApiController.class);
+  private final LandingZoneService landingZoneService;
+  private final FeatureConfiguration features;
+  private final CrlService crlService;
+  private final AzureConfiguration azureConfiguration;
+
+  @Autowired
+  public AzureLandingZoneApiController(
+      AuthenticatedUserRequestFactory authenticatedUserRequestFactory,
+      HttpServletRequest request,
+      SamService samService,
+      LandingZoneService landingZoneService,
+      CrlService crlService,
+      AzureConfiguration azureConfiguration,
+      FeatureConfiguration features) {
+    super(authenticatedUserRequestFactory, request, samService);
+    this.landingZoneService = landingZoneService;
+    this.crlService = crlService;
+    this.azureConfiguration = azureConfiguration;
+    this.features = features;
+  }
+
+  @Override
+  public ResponseEntity<ApiCreatedAzureLandingZoneResult> createAzureLandingZone(
+      @RequestBody ApiCreateAzureLandingZoneRequestBody body) {
+    features.isAzureEnabled();
+    String landingZoneDetails = "definition='%s', version='%s'";
+    logger.info(
+        "Requesting new Azure landing zone with the following parameters: {}",
+        String.format(landingZoneDetails, body.getDefinition(), body.getVersion()));
+
+    ApiAzureContext azureContext =
+        Optional.ofNullable(body.getAzureContext())
+            .orElseThrow(
+                () ->
+                    new CloudContextRequiredException(
+                        "AzureContext is required when creating an Azure landing zone"));
+
+    LandingZoneRequest landingZoneRequest =
+        LandingZoneRequest.builder()
+            .definition(body.getDefinition())
+            .version(body.getVersion())
+            .parameters(
+                MapperUtils.LandingZoneMapper.landingZoneParametersFrom(body.getParameters()))
+            .azureCloudContext(MapperUtils.AzureCloudContextMapper.from(azureContext))
+            .build();
+    String jobId =
+        landingZoneService.startLandingZoneCreationJob(
```

Comment:
I would like to see this split into a controller and a "dispatch", as I did with TPS, so that WSM can call the dispatch layer using the API objects. That makes it simple to retarget the WSM code to call LZ if/when it is its own service.

Otherwise, WSM "knows" about the library interface instead of the REST interface and the lift is larger later on.